base_layer/wallet/src/error.rs

```diff
@@ -173,6 +173,75 @@ impl From<WalletStorageError> for ExitCodes {
 
 impl PartialEq for WalletStorageError {
     fn eq(&self, other: &Self) -> bool {
-        self == other
+        match other {
```

Comment:
I definitely don't think this is needed. Rather just don't implement PartialEq. The reason is that anytime anyone adds a new error to this that derives from an error in another crate that doesn't implement PartialEq it will break this. We do not want to maintain this. You will see many errors in the system do not have this because its such a nightmare.

Anywhere that wants to check if the WalletStorageError equals something then they can use matches!() themselves.

Reply:
Ok, removing `PartialEq` from dependent errors.